Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPITestCocoa.mm

```diff
@@ -113,6 +115,19 @@
     return *m_onMessage;
 }
 
+JSValue *WebExtensionAPITest::runWithUserGesture(WebFrame& frame, JSValue *function)
+{
+    RefPtr coreFrame = frame.protectedCoreLocalFrame();
+    WebCore::UserGestureIndicator gestureIndicator(WebCore::IsProcessingUserGesture::Yes, coreFrame ? coreFrame->document() : nullptr);
+
+    return [function callWithArguments:@[ ]];
```

Comment:
Will this only work if the function is synchronous?

Reply:
Async functions return a promise, so it works with either. Our tests do pass asynchronous functions here for some cases.